File: bench/realworld/bench_strbuild.js
// bench_strbuild.js (qjs) - build a large string by naive s += piece, N times.
var N = 1000000;
var t0 = Date.now();
var s = "";
for (var i = 0; i < N; i++) s += "item_" + i + ";";
var result = s.length;
var elapsed = Date.now() - t0;
print("result: " + result);
print("elapsed: " + elapsed + " ms");

File: bench/realworld/bench_strbuild.go
// bench_strbuild.go - build a large string, N times, via strings.Builder
// (Go's idiomatic linear builder; naive s += is O(n^2) because strings are
// immutable). result is the final byte length.
package main

import (
	"fmt"
	"strconv"
	"strings"
	"time"
)

func main() {
	N := 1000000
	t0 := time.Now()
	var sb strings.Builder
	for i := 0; i < N; i++ {
		sb.WriteString("item_")
		sb.WriteString(strconv.Itoa(i))
		sb.WriteString(";")
	}
	result := sb.Len()
	elapsed := time.Since(t0).Milliseconds()
	fmt.Printf("result: %d\n", result)
	fmt.Printf("elapsed: %d ms\n", elapsed)
}
